impl/src/main/java/org/jboss/weld/util/Observers.java

```diff
@@ -84,7 +85,8 @@ private Observers() {
 
     public static boolean isContainerLifecycleObserverMethod(ObserverMethod<?> method) {
         return CONTAINER_LIFECYCLE_EVENT_TYPES.contains(Reflections.getRawType(method.getObservedType()))
-                || (Object.class.equals(method.getObservedType()) && method instanceof ContainerLifecycleEventObserverMethod);
+                || (Object.class.equals(method.getObservedType()) && method.getObservedQualifiers().size() == 1
```

Comment:
I see. Well, when I adjusted the condition to accept this observer as well, I am getting a `DefinitionException` saying that CLE observer cannot inject anything else than BM. I'll take a closer look into it.